llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -789,9 +791,24 @@ int main(int argc, char **argv) {
   if (OutputFilename.getNumOccurrences() == 0)
     OutputFilename = (Twine(sys::path::stem(InputFilename)) + ".files").str();
 
+  TableFiles TblFiles = processOneModule(std::move(M));
   util::SimpleTable Table;
-  int Res = processOneModule(std::move(M), Table);
-  if (Res)
+
+  auto addTableColumn = [&Table, &TblFiles](std::string Str) {
```

Comment:
`std::string` -> `const std::string &`: I guess there is no need to copy that string index every time this lambda is called

Reply:
I think there is no additional copy. The `std:string` object is constructed from raw string and then moved to the local `Str` parameter.

Reply:
> The `std:string` object is constructed from raw string and then moved to the local `Str` parameter.

If I understand correctly, `std::string` creates a copy of that raw string: [cppreference](https://en.cppreference.com/w/cpp/string/basic_string/basic_string), see constructor (5):

>  Constructs the string with the contents initialized with a copy of the null-terminated character string pointed to by s. The length of the string is determined by the first null character. The behavior is undefined if [s, s + Traits::length(s)) is not a valid range (for example, if s is a null pointer). This constructor is not used for class template argument deduction if the Allocator type that would be deduced does not qualify as an allocator. (since C++17)

Reply:
> > The `std:string` object is constructed from raw string and then moved to the local `Str` parameter.
> 
> If I understand correctly, `std::string` creates a copy of that raw string: [cppreference](https://en.cppreference.com/w/cpp/string/basic_string/basic_string), see constructor (5):
> 
> > Constructs the string with the contents initialized with a copy of the null-terminated character string pointed to by s. The length of the string is determined by the first null character. The behavior is undefined if [s, s + Traits::length(s)) is not a valid range (for example, if s is a null pointer). This constructor is not used for class template argument deduction if the Allocator type that would be deduced does not qualify as an allocator. (since C++17)

Well, yes. But changing `std::string` -> `const std::string &` in `addTableColumn` signature will not eliminate constructing a `std::string` object from raw string, right. So, there will be one `string` object constructed anyway. Then it will be either moved (if passed by value) or passed by reference.

Does it make sense or maybe I misunderstood your comment?

Reply:
> Does it make sense or maybe I misunderstood your comment?

Yes, it makes sense. I've just realized that there are no other copies of that string and changing `string` -> `const string &` indeed won't eliminate it initial construction, my bad